core/Db/Adapter.php

```diff
@@ -39,7 +39,13 @@ public static function factory($adapterName, & $dbInfos, $connect = true)
 
         $className = self::getAdapterClassName($adapterName);
 
-        $adapter   = new $className($dbInfos);
+        // make sure not to pass any references otherwise they will modify $dbInfos
+        $infos = array();
+        foreach ($dbInfos as $key => $val) {
+            $infos[$key] = $val;
+        }
+
```

Comment:
each time a new database instance was created we ended up having eg instead of `127.0.0.1` the next time `host=127.0.0.1`, the next time `host=host=127.0.0.1`, ...